runtime/worker_sleep.h

```diff
@@ -99,30 +99,20 @@ static void swap_worker_with_target(global_state *g, worker_id self,
     worker_to_index[self] = target_index;
 }
 
+// These functions return the old value
+
 __attribute__((always_inline)) static inline uint64_t
 add_to_sentinels(global_state *const rts, int32_t val) {
+    // val is sign extended to 64 bits
     return atomic_fetch_add_explicit(&rts->disengaged_sentinel, val,
                                      memory_order_release);
 }
 
 __attribute__((always_inline)) static inline uint64_t
 add_to_disengaged(global_state *const rts, int32_t val) {
-    while (true) {
-        uint64_t disengaged_sentinel = atomic_load_explicit(
-            &rts->disengaged_sentinel, memory_order_relaxed);
-        uint32_t disengaged = GET_DISENGAGED(disengaged_sentinel);
-        uint32_t sentinel = GET_SENTINEL(disengaged_sentinel);
-        uint64_t new_disengaged_sentinel =
-            DISENGAGED_SENTINEL(disengaged + val, sentinel);
-
-        if (atomic_compare_exchange_strong_explicit(
-                &rts->disengaged_sentinel, &disengaged_sentinel,
-                new_disengaged_sentinel, memory_order_release,
-                memory_order_acquire))
-            return disengaged_sentinel;
-
-        busy_loop_pause();
-    }
+    return atomic_fetch_add_explicit(&rts->disengaged_sentinel,
+                                     (uint64_t)val << 32,
```

Comment:
Can we avoid the constant `32` in the code here?  Something like `DISENGAGED_SENTINEL(val, 0)` might work, instead of `(uint64_t)val << 32`, while being easier to understand and maintain.